pkg/controller/node/nodecontroller_test.go

```diff
@@ -1265,39 +1265,48 @@ func TestMonitorNodeStatusEvictPodsWithDisruption(t *testing.T) {
 		if err := nodeController.monitorNodeStatus(); err != nil {
 			t.Errorf("%v: unexpected error: %v", item.description, err)
 		}
-		// Give some time for rate-limiter to reload
-		time.Sleep(500 * time.Millisecond)
-
 		for zone, state := range item.expectedFollowingStates {
 			if state != nodeController.zoneStates[zone] {
 				t.Errorf("%v: Unexpected zone state: %v: %v instead %v", item.description, zone, nodeController.zoneStates[zone], state)
 			}
 		}
-		zones := testutil.GetZones(fakeNodeHandler)
-		for _, zone := range zones {
-			// Time for rate-limiter reloading per node.
-			time.Sleep(50 * time.Millisecond)
-			nodeController.zonePodEvictor[zone].Try(func(value TimedValue) (bool, time.Duration) {
-				uid, _ := value.UID.(string)
-				deletePods(fakeNodeHandler, nodeController.recorder, value.Value, uid, nodeController.daemonSetStore)
-				return true, 0
-			})
-		}
-
-		podEvicted := false
-		for _, action := range fakeNodeHandler.Actions() {
-			if action.GetVerb() == "delete" && action.GetResource().Resource == "pods" {
-				podEvicted = true
+		podEvicted := new(bool)
+		start := time.Now()
+		// Infinite loop, used for retrying in case ratelimiter fails to reload for Try function.
+		// this breaks when we have the status that we need for test case or when we don't see the
+		// intended result after 1 minute.
+		for {
+			nodeController.doEviction(fakeNodeHandler, podEvicted)
+			if *podEvicted == item.expectedEvictPods || time.Since(start) > 1*time.Minute {
 				break
 			}
 		}
 
-		if item.expectedEvictPods != podEvicted {
+		if item.expectedEvictPods != *podEvicted {
 			t.Errorf("%v: expected pod eviction: %+v, got %+v", item.description, item.expectedEvictPods, podEvicted)
 		}
 	}
 }
 
+// doEviction does the fake eviction and returns the status of eviction operation.
+func (nc *nodeController) doEviction(fakeNodeHandler *testutil.FakeNodeHandler, podEvicted *bool) {
```

Comment:
Yup, you are right. I did not want to create another variable, so I am passing the ref around. I am fine with returning a bool as well. I will make that change.